### ISSUE TITLE
Support for --exit-node=X.X.X.X

### ISSUE THREAD

**Issue by michael-staffa** [Fluent-networks/tailscale-mikrotik#16]
Hello!

I have a use-case for all traffic in a subnet to pass out through an exit node I have running on my Tailnet.
I know it's possible to configure a single user device to route over an exit node using `--exit-node=X.X.X.X` but I'm not sure how to accomplish the same on my MikroTik router.

Your newest release includes the ability to specify extra arguments, and there is a section in the `tailscale.sh` script that enables routes for returning packets:
```
# Install routes
IFS=',' read -ra SUBNETS <<< "${ADVERTISE_ROUTES}"
for s in "${SUBNETS[@]}"; do
  ip route add "$s" via "${CONTAINER_GATEWAY}"
done
```

What I want to know is how to route outbound packets to the Tailscale container and have them leave/enter through my exit node.
Is this something you can provide a suggestion on?

**Comment by frankedwards** [comment]
Hi Michael,

I don't think it's possible for traffic from non-Tailscale hosts to use an exit node via an intermediate host/router.

Whilst what you're suggesting with regard to ```--exit-node=x.x.x.x``` above is possible, I believe the exit node would only be accessible for traffic originating from the authenticated container.

**Comment by michael-staffa** [comment]
Hi @frankedwards

Would it be possible to NAT traffic through the subnet router's tailscale IP?

For instance, if I have a MikroTik router with a subnet `10.1.0.0/24` that routes outbound traffic to `0.0.0.0/0` through the container (as a gateway) are there IPtables rules that can be set up inside the container that NAT it out on the Tailscale interface/IP?

Does all traffic _have_ to originate from the Tailscale network in order for return traffic to be accepted?

**Comment by frankedwards** [comment]
Hi Michael

Perhaps... have a look at this article... it should be possible to shell into the container to test the iptables config.
https://forum.tailscale.com/t/sending-traffic-through-exit-node/4307

**Comment by michael-staffa** [comment]
I think there's a problem with a missing module in the kernel that MikroTik is using. The available iptables module is out of date and I can't change any rules as far as I can tell.

**Comment by frankedwards** [comment]
This appears to work for me.

```
bash-5.1# iptables -t nat -L
Chain PREROUTING (policy ACCEPT)
target     prot opt source               destination

Chain INPUT (policy ACCEPT)
target     prot opt source               destination

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination

Chain POSTROUTING (policy ACCEPT)
target     prot opt source               destination
ts-postrouting  all  --  anywhere             anywhere
MASQUERADE  all  --  anywhere             anywhere
MASQUERADE  all  --  anywhere             anywhere
MASQUERADE  all  --  anywhere             anywhere
           all  --  anywhere             anywhere
MASQUERADE  all  --  anywhere             anywhere

Chain ts-postrouting (1 references)
target     prot opt source               destination
bash-5.1# iptables -t nat -L
Chain PREROUTING (policy ACCEPT)
target     prot opt source               destination

Chain INPUT (policy ACCEPT)
target     prot opt source               destination

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination

Chain POSTROUTING (policy ACCEPT)
target     prot opt source               destination
ts-postrouting  all  --  anywhere             anywhere
MASQUERADE  all  --  anywhere             anywhere
MASQUERADE  all  --  anywhere             anywhere
MASQUERADE  all  --  anywhere             anywhere
           all  --  anywhere             anywhere
MASQUERADE  all  --  anywhere             anywhere

Chain ts-postrouting (1 references)
target     prot opt source               destination
bash-5.1# iptables -t nat -A POSTROUTING -s 10.1.0.0/24 -o tailscale0 -j MASQUERADE
bash-5.1# iptables -t nat -L
Chain PREROUTING (policy ACCEPT)
target     prot opt source               destination

Chain INPUT (policy ACCEPT)
target     prot opt source               destination

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination

Chain POSTROUTING (policy ACCEPT)
target     prot opt source               destination
ts-postrouting  all  --  anywhere             anywhere
MASQUERADE  all  --  anywhere             anywhere
MASQUERADE  all  --  anywhere             anywhere
MASQUERADE  all  --  anywhere             anywhere
           all  --  anywhere             anywhere
MASQUERADE  all  --  anywhere             anywhere
MASQUERADE  all  --  10.1.0.0/24          anywhere

Chain ts-postrouting (1 references)
target     prot opt source               destination
```

**Comment by michael-staffa** [comment]
Many thanks @frankedwards!
That seems to have done it! 

After much head-scratching we figured out that the iptables problem is with the cloud-hosted RouterOS image. As soon as we switched to a live test router it began working exactly as expected.

If it's alright with you I'd like to write up a PR at not-2-oclock-in-the-morning that can handle exit node settings if they're passed in.

Thanks again!!
